app/src/main/java/com/github/houseorganizer/houseorganizer/panels/billsharer/BalanceActivity.java

```diff
@@ -55,6 +59,8 @@ private void initializeData(){
                             view.setLayoutManager(linearLayoutManager);
                             view.setAdapter(adapter);
                         });
+
+                        LocalStorage.pushDebtsOffline(getApplicationContext(), currentHouse.getId(), bs.getDebts());
```

Comment:
You should put it inside the complete listener otherwise the billsharer will not be complete, same for expense

Reply:
Right, my bad. I thought the billsharer was ready in the first listener. Changes coming up!